torch/csrc/jit/codegen/cuda/ir_base_nodes.cpp

```diff
@@ -21,9 +24,11 @@ namespace jit {
 namespace fuser {
 namespace cuda {
 
-Statement::Statement(IrBuilderPasskey) {}
+Statement::Statement(IrBuilderPasskey passkey)
+    : is_kir_stmt_(passkey.ir_container_ != nullptr ? false : true) {}
 
-Statement::Statement(const Statement* src, IrCloner* ir_cloner) {
+Statement::Statement(const Statement* src, IrCloner* ir_cloner)
+    : is_kir_stmt_(false) {
```

Comment:
Is it assumed no KIR stmt is cloned?

Reply:
Yes, IrCloner generally does not support kir, this will be changed/fixed when Kernel is based on IrContainer in some way.